### PR TITLE
refactor: better gitbook script, organise assets better and more documentation

### PR DESCRIPTION
**Pull request by jeremybarbet**
## What

- Add whitelist for full uppercase words (e.g. API, CMS, etc..)
- Add an exception for CONNECTIVES word to be capitalized if it's the first word of the title
- Remove `.gitbook/assets` directory and move assets to the directory where it's used under an `assets` directory
- Add documentation about README.md, links, and assets
- Fix existing documentation to render properly links and structure

### CODE REVIEW

**Review by sindrig — DISMISSED** [review]
LGTM

File: scripts/gitbook.js [hunk 1]
```diff
@@ -13,6 +13,8 @@ const ROOT_LEVEL = 3
  * Format names following https://en.wikipedia.org/wiki/AP_Stylebook and https://en.wikipedia.org/wiki/APA_style styles
  */
 const CONNECTIVES = 'a an and at but by for in nor of on or so the to up yet'
+const WHITELIST = 'api css cms adr http nova sms ui'
```
Comment (darrikonn) [inline scripts/gitbook.js:16]: `UPPERCASE_WHITELIST`

Had to read the code to see what it was a whitelist for.

**Review by darrikonn — CHANGES_REQUESTED** [review]
Looks good!

**Review by darrikonn — APPROVED** [review]

**Comment by sindrig** [comment]
formatting seems to have some problems with SUMMARY.md now: https://github.com/island-is/island.is/pull/2443/checks?check_run_id=1680227716

**Review by darrikonn — APPROVED** [review]

**Pull request merged**